metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlTable.java

```diff
@@ -0,0 +1,217 @@
+        } finally {
+            this.fastServiceMetric.recordTimer(
+                HiveMetrics.TagTableExists.getMetricName(), registry.clock().wallTime() - start);
+        }
+        return result;
+    }
+
+    /**
+     * Returns all the table names referring to the given <code>uris</code>.
+     *
+     * @param uris         locations
+     * @param prefixSearch if true, we look for tables whose location starts with the given <code>uri</code>
+     * @return map of uri to list of partition names
+     * @throws UnsupportedOperationException If the connector doesn't implement this method
+     */
+    @Transactional(readOnly = true)
+    public Map<String, List<QualifiedName>> getTableNames(final List<String> uris, final boolean prefixSearch) {
+        final long start = registry.clock().wallTime();
+        // Create the sql
+        final StringBuilder queryBuilder = new StringBuilder(SQL.GET_TABLE_NAMES_BY_URI);
+        final List<SqlParameterValue> params = Lists.newArrayList();
+        if (prefixSearch) {
+            queryBuilder.append(" and (1=0");
+            uris.forEach(uri -> {
+                queryBuilder.append(" or location like ?");
+                params.add(new SqlParameterValue(Types.VARCHAR, uri + "%"));
+            });
+            queryBuilder.append(" )");
+        } else {
+            queryBuilder.append(" and location in (");
+            uris.forEach(uri -> {
+                queryBuilder.append("?,");
+                params.add(new SqlParameterValue(Types.VARCHAR, uri));
+            });
+            queryBuilder.deleteCharAt(queryBuilder.length() - 1).append(")");
+        }
+        ResultSetExtractor<Map<String, List<QualifiedName>>> handler = rs -> {
+            final Map<String, List<QualifiedName>> result = Maps.newHashMap();
+            while (rs.next()) {
+                final String schemaName = rs.getString("schema_name");
+                final String tableName = rs.getString("table_name");
+                final String uri = rs.getString("location");
+                final List<QualifiedName> names = result.computeIfAbsent(uri, k -> Lists.newArrayList());
+                names.add(QualifiedName.ofTable(catalogName, schemaName, tableName));
+            }
+            return result;
+        };
+        try {
+            return jdbcTemplate.query(queryBuilder.toString(), params.toArray(), handler);
+        } finally {
+            this.fastServiceMetric.recordTimer(
+                HiveMetrics.TagGetTableNames.getMetricName(), registry.clock().wallTime() - start);
+        }
+    }
+
+    /**
+     * Locks the iceberg table for update so that no other request can modify the table.
+     * @param tableId table internal id
+     */
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    public void lockIcebergTable(final Long tableId) {
+        Boolean isLocked = null;
+        try {
+            isLocked = jdbcTemplate.queryForObject(SQL.ICEBERG_TABLE_LOCK,
```

Comment:
The sql here is to lock a table parameter row, which is quite generic. The sql shall reflect what this is doing. So instead of saying ICEBERG_TABLE_LOCK, we can say TABLE_PARAM_LOCK. We can still leave the function name as lockIcebergTable to be specific and explicit.

Reply:
Will change.